core/src/main/scala/chisel3/internal/Builder.scala

```diff
@@ -791,13 +794,35 @@ private[chisel3] object Builder extends LazyLogging {
 
   def currentWhen: Option[WhenContext] = dynamicContext.whenStack.headOption
 
-  def currentClock: Option[Clock] = dynamicContext.currentClock
-  def currentClock_=(newClock: Option[Clock]): Unit = {
+  // Helper for reasonable errors when clock or reset value not yet initialized
+  private def getDelayed[A](field: String, dc: Delayed[A]): A = {
+    val result = dc.value
+    if (result == null) {
+      // TODO add SourceInfo and change to Builder.exception
+      throwException(
+        s"The implicit $field is null which means its definition probably has not yet been initialized."
```

Comment:
```suggestion
        s"The implicit $field is null which means the code that sets its definition has not yet executed."
```

maybe? This is an error in execution order that I can certainly imagine people running into.

Reply:
Yeah I think that phrasing will be better for most users (even if the "key phrase" to Google for is "Scala initialization order")